src/Raven.Server/Rachis/ClusterMaintenanceSlave.cs

```diff
@@ -0,0 +1,135 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Raven.Client.Documents.Replication.Messages;
+using Raven.Server.Documents.TcpHandlers;
+using Raven.Server.ServerWide;
+using Raven.Server.ServerWide.Context;
+using Sparrow;
+using Sparrow.Json;
+using Sparrow.Json.Parsing;
+
+namespace Raven.Server.Rachis
+{
+    public class ClusterMaintenanceSlave : IDisposable
+    {
+        private readonly TcpConnectionOptions _tcp;
+        private readonly ServerStore _server;
+        private CancellationToken _token;
+        private readonly CancellationTokenSource _cts;
+        private Task _collectingTask;
+        private Dictionary<string,DatabaseStatusReport> _cachedReport = new Dictionary<string, DatabaseStatusReport>();
```

Comment:
What about when a database is unloaded. We want to keep it's last known statistics and send them over to the leader.
We can't keep this info on the leader, because the leadership can change.